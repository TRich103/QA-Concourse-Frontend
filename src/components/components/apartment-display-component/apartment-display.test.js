import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme,{shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ApartmentDisplay from './apartment-display.component';
import toJson from 'enzyme-to-json';
import apartment1 from './apartment.jpg'
import { Media } from 'reactstrap';

Enzyme.configure({adapter: new Adapter()})

describe('Apartment Display component', () =>{

  var heading = "Test Heading"
  var body = "Test Body"
  var img = apartment1

  //Creates snapshot of desired display
  test('renders desired display', () => {
    var imgStyle = {
        maxWidth: "256px",
        maxHeight: "256px",
        paddingRight : "10%",
        paddingTop: "1%"
      };
      
      var padding ={
        padding: "1%"
      }

      const display = mount(
        <div style={padding}>
            <Media>
            <Media left href="#">
            <Media style={imgStyle} object src={img}/>
            </Media>
            <Media body>
            <Media heading>
                {heading}
            </Media>
                {body}
            </Media>
        </Media>
        </div>
      )
    expect(toJson(display)).toMatchSnapshot()
  })

  //Checks if QATable component matches desired table
  test('Apartment display matches desired outcome', () => {
    const display = mount(
        <ApartmentDisplay
            heading = {heading}
            img = {apartment1}
            body = {body}
        />
    )
    expect(toJson(display)).toMatchSnapshot()
  })

})