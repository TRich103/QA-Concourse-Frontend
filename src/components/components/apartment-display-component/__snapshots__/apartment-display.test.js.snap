// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apartment Display component Apartment display matches desired outcome 1`] = `
<ApartmentList
  body="Test Body"
  heading="Test Heading"
  img="apartment.jpg"
>
  <div
    style={
      Object {
        "padding": "1%",
      }
    }
  >
    <Media>
      <div
        className="media"
      >
        <Media
          href="#"
          left={true}
        >
          <a
            className="media-left"
            href="#"
          >
            <Media
              object={true}
              src="apartment.jpg"
              style={
                Object {
                  "maxHeight": "256px",
                  "maxWidth": "256px",
                  "paddingRight": "10%",
                  "paddingTop": "1%",
                }
              }
            >
              <img
                className="media-object"
                src="apartment.jpg"
                style={
                  Object {
                    "maxHeight": "256px",
                    "maxWidth": "256px",
                    "paddingRight": "10%",
                    "paddingTop": "1%",
                  }
                }
              />
            </Media>
          </a>
        </Media>
        <Media
          body={true}
        >
          <div
            className="media-body"
          >
            <Media
              heading={true}
            >
              <h4
                className="media-heading"
              >
                Test Heading
              </h4>
            </Media>
            Test Body
          </div>
        </Media>
      </div>
    </Media>
  </div>
</ApartmentList>
`;

exports[`Apartment Display component renders desired display 1`] = `
<div
  style={
    Object {
      "padding": "1%",
    }
  }
>
  <Media>
    <div
      className="media"
    >
      <Media
        href="#"
        left={true}
      >
        <a
          className="media-left"
          href="#"
        >
          <Media
            object={true}
            src="apartment.jpg"
            style={
              Object {
                "maxHeight": "256px",
                "maxWidth": "256px",
                "paddingRight": "10%",
                "paddingTop": "1%",
              }
            }
          >
            <img
              className="media-object"
              src="apartment.jpg"
              style={
                Object {
                  "maxHeight": "256px",
                  "maxWidth": "256px",
                  "paddingRight": "10%",
                  "paddingTop": "1%",
                }
              }
            />
          </Media>
        </a>
      </Media>
      <Media
        body={true}
      >
        <div
          className="media-body"
        >
          <Media
            heading={true}
          >
            <h4
              className="media-heading"
            >
              Test Heading
            </h4>
          </Media>
          Test Body
        </div>
      </Media>
    </div>
  </Media>
</div>
`;
